### PR TITLE
Fix web worker in watch mode

### PR DESCRIPTION
**Pull request by birkskyum**
Currently, the web worker is only enabled if watch mode is off, because the rollup node-resolve plugin doesn't look at typescript files by default, and it was configured to the wrong path. This pr fixes that, and it can hopefully unblock #961 
To test this PR, first go to the main branch and:

1. Delete the rollup/build/tsc/src folder
2. Run `npm run start-debug`
3. Open chrome dev console, and go to "sources" tap - there will be just "top -> localhost -> local files" in the left pane, which indicates that the web worker ain't running.

Now checkout this pr, and do the same.

The sources tap should now have "top" and some web workers, like in the image

Screenshot from 2022-02-08 20-46-32.

### CODE REVIEW

File: bench/rollup_config_benchmarks.ts [hunk 1]
```diff
@@ -67,7 +67,11 @@ const viewConfig: RollupOptions = {
         sourcemap: false
     },
     plugins: [
-        resolve({browser: true, preferBuiltins: false}),
+        resolve({
+            browser: true,
```
Comment (HarelM) [inline bench/rollup_config_benchmarks.ts:71]: These lines send to be repeated 3 times in 3 different files.
Rollup_plugins file should be a centralized place where these kind of things are there for other config file to use, I believe.

File: build/rollup_plugins.ts [hunk 2]
```diff
@@ -40,7 +40,8 @@ export const plugins = (minified: boolean, production: boolean, watch: boolean):
     }) : false,
     resolve({
         browser: true,
-        preferBuiltins: false
+        preferBuiltins: false,
+        extensions: ['.mjs', '.js', '.json', '.node', '.ts']
```
Comment (HarelM) [inline build/rollup_plugins.ts:44]: Can we remove the watch flag or will this be a different PR?

**Comment by HarelM** [comment]
Nice work! Keep it up!

File: bench/rollup_config_benchmarks.ts (hunk 1 above)
Comment (birkskyum) [reply bench/rollup_config_benchmarks.ts]: I actually just realized that we only need to make this change in the one file rollup.config.ts, because benchmarks and style-spec for now.

File: build/rollup_plugins.ts (hunk 2 above)
Comment (birkskyum) [reply build/rollup_plugins.ts:44]: I haven't investigated this, and don't know what the consequence would be.

File: bench/rollup_config_benchmarks.ts (hunk 1 above)
Comment (HarelM) [reply bench/rollup_config_benchmarks.ts]: Seems weird, why would style spec and benchmarks not need to replace the worker mock implementation?

File: build/rollup_plugins.ts (hunk 2 above)
Comment (HarelM) [reply build/rollup_plugins.ts:44]: Ok, let's leave it out of this PR and address this in a different one.

File: bench/rollup_config_benchmarks.ts (hunk 1 above)
Comment (birkskyum) [reply bench/rollup_config_benchmarks.ts:71]: I reverted so they replace as well - I just wasn't sure if it was necessary, and didn't want to make too many changes.

Comment (birkskyum) [reply bench/rollup_config_benchmarks.ts:71]: But now I see that when benchmark rollup is updated, the workers appear there as well.
bench

Comment (HarelM) [reply bench/rollup_config_benchmarks.ts:71]: I think it should be there, shouldn't it?
If you run the benchmark before these changes you don't see the workers?

Comment (birkskyum) [reply bench/rollup_config_benchmarks.ts:71]: Correct

**Comment by birkskyum** [comment]
The benchmarks are working now because of #969

File: bench/rollup_config_benchmarks.ts (hunk 1 above)
Comment (HarelM) [reply bench/rollup_config_benchmarks.ts:71]: The code duplication here is still not addressed as far as I understand.

Comment (birkskyum) [reply bench/rollup_config_benchmarks.ts]: I made it such that the configuration only exist in rollup_plugins.ts now, and export where needed.

**Review by HarelM — APPROVED** [review]
Perfect, Thanks!
